### DISCUSSION TITLE
Connectors with Service Principals - Cannot update secret

### DISCUSSION THREAD

**Discussion by StylezDevops** [microsoft/powerplatform-build-tools#452] [Ideas]
ENHANCEMENT REQUEST:
@kkanakas
If one creates a connector using a Service Principal (eg: Dataverse) there is currently no method of changing the ClientSecret when it expires. This means we have to create a brand new connector and then update all the connection references that use this connector. 

On large complex projects this is onerous and challenging.  We would like to see this behaviour addressed sooner rather than later as it kind of makes the feature useless.
Kind Regards
Dan Styles

**Comment by jboliveira** [comment]
Furthermore, utilizing Managed Environments, we may do a self-discovery of Service Principals accessible for that specific environment and then simply choose the one necessary for the connection, eliminating the need to update it in the event of a service principal rotation.

It may also be a PAC CLI feature where we can use the service principal to make new connections and then automate the replacement of the `connectionId` in `deploymentSettings.json`. 

Continue: #481

**Comment by jbuenoliveira** [comment]
This feature is generally available (GA) and already covers Dataverse connectors. Is there something specific that you find missing?
